src/facade/dragonfly_connection.cc

```diff
@@ -1518,6 +1526,20 @@ void Connection::SquashPipeline() {
 
   cc_->async_dispatch = true;
 
+  // We need to first set async_dispatch before we preempt. Otherwise, when the AsyncFiber
+  // wakes up, sync_dispatch might be true, violating the precondition of this flow
+  // (when we block earlier in the body of AsyncFiber at cnd_.wait(noop_lk, [this])...)
+  fb2::NoOpLock noop;
```

Comment:
still I dont see you use QueueBackpressure class today. Why not use the same class for throttle on squashed pipeline reply size?

Reply:
ok, you are right, squashing_current_reply_size can be higher than a threshold due to other connections on this thread. In that case I suggest that instead of throttling the connection - we just avoid calling SquashPipeline, and choose the usual route of sending a single command (lines 1644-1647)

Reply:
Yeah and that would remove the throttling/extra synchronization all together +1